app/models/mixins/inter_region_api_method_relay.rb

```diff
@@ -0,0 +1,87 @@
+module InterRegionApiMethodRelay
+  def self.extended(klass)
+    unless klass.const_defined?("InstanceMethodRelay")
+      instance_relay = klass.const_set("InstanceMethodRelay", Module.new)
+      klass.prepend(instance_relay)
+    end
+
+    unless klass.const_defined?("ClassMethodRelay")
+      class_relay = klass.const_set("ClassMethodRelay", Module.new)
+      klass.singleton_class.prepend(class_relay)
+    end
+  end
+
+  def api_relay_method(method, action = method)
+    relay = const_get("InstanceMethodRelay")
+    collection_name = collection_for_class
```

Comment:
So because `collection_for_class` uses `self` to determine what collection we are using, it has to be called when `self` is the extending class rather than the relay module.

Calling this here allows us to use collection_name in the `defind_method` because of the closure that is created.

I agree it does look weird. I'm all for a different way of doing it, but if not maybe a comment is justified?